commcare_connect/data_export/serializer.py

```
@@ -435,3 +479,167 @@ class Meta:
+        qs = WorkAreaGroup.objects.filter(opportunity=opportunity, name=value)
+        if self.instance:
+            qs = qs.exclude(pk=self.instance.pk)
+        if qs.exists():
+            raise serializers.ValidationError(_("A work area group with this name already exists."))
+        return value
+
+    def create(self, validated_data):
+        try:
+            return WorkAreaGroup.objects.create(opportunity=self.context["view"].opportunity, **validated_data)
+        except IntegrityError:
+            raise serializers.ValidationError({"name": _("A work area group with this name already exists.")})
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)